modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseModelListenerCheck.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.poshi.core.util.StringPool;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+import com.liferay.source.formatter.parser.JavaClass;
+import com.liferay.source.formatter.parser.JavaClassParser;
+import com.liferay.source.formatter.parser.JavaMethod;
+import com.liferay.source.formatter.parser.JavaParameter;
+import com.liferay.source.formatter.parser.JavaSignature;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeJavaBaseModelListenerCheck extends BaseUpgradeCheck {
+
+	@Override
+	protected String format(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		JavaClass javaClass = JavaClassParser.parseJavaClass(fileName, content);
+
+		List<String> implementedClassNames =
+			javaClass.getImplementedClassNames();
+
+		if (implementedClassNames.contains("BaseModelListener")) {
+			return content;
+		}
+
+		for (JavaTerm childJavaTerm : javaClass.getChildJavaTerms()) {
+			if (!childJavaTerm.isJavaMethod()) {
+				continue;
+			}
+
+			JavaMethod javaMethod = (JavaMethod)childJavaTerm;
+
+			String javaMethodContent = javaMethod.getContent();
+
+			String newJavaMethodContent = _checkMethodDefinition(
+				javaMethodContent, javaMethod, _onAfterUpdatePattern);
+
+			newJavaMethodContent = _checkMethodDefinition(
+				newJavaMethodContent, javaMethod, _onfBeforeUpdatePattern);
+
+			content = StringUtil.replace(
+				content, javaMethodContent, newJavaMethodContent);
+		}
+
+		return content;
+	}
+
+	private String _checkMethodDefinition(
+		String javaMethodContent, JavaMethod javaMethod, Pattern pattern) {
```

Comment:
@michaelmdelcavalcanti, with one pattern for both cases, we don't need a parameter for the pattern and can use it directly.